### PR TITLE
Fix blogging module post detail page permission check.

### LINKED ISSUE #3098: Blogging post detail page may have wrong creator authority check.
https://github.com/abpframework/abp/blob/42f37c5ff01ad853a5425d15539d4222cd0dab69/modules/blogging/src/Volo.Blogging.Application/Volo/Blogging/Posts/PostAuthorizationHandler.cs#L36-L59

https://github.com/abpframework/abp/blob/42f37c5ff01ad853a5425d15539d4222cd0dab69/modules/blogging/src/Volo.Blogging.Application/Volo/Blogging/Comments/CommentAuthorizationHandler.cs#L36-L59

It seems that users cannot delete their own posts/comments, but in the detail page I found:

https://github.com/abpframework/abp/blob/42f37c5ff01ad853a5425d15539d4222cd0dab69/modules/blogging/src/Volo.Blogging.Web/Pages/Blog/Posts/Detail.cshtml#L70-L83

https://github.com/abpframework/abp/blob/42f37c5ff01ad853a5425d15539d4222cd0dab69/modules/blogging/src/Volo.Blogging.Web/Pages/Blog/Posts/Detail.cshtml#L235-L241

### PR DESCRIPTION
**Pull request by gdlcf88**
Fix #3098

### CODE REVIEW

**Comment by maliming** [comment]
hi @gdlcf88 

### Conflicting files
`modules/blogging/src/Volo.Blogging.Web/Pages/Blog/Posts/Detail.cshtml`

**Comment by gdlcf88** [comment]
I will do it.

**Comment by gdlcf88** [comment]
Done.

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 1]
```diff
@@ -57,14 +57,14 @@
...
+                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Posts.Update) || (CurrentUser.Id.HasValue && CurrentUser.Id == Model.Post.CreatorId))
                                                 {
                                                     <span class="seperator">|</span>
                                                     <a asp-page="./Edit" asp-route-postId="@Model.Post.Id" asp-route-blogShortName="@Model.BlogShortName">
                                                         <i class="fa fa-pencil"></i> @L["Edit"]
                                                     </a>
                                                 }
-                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Posts.Delete) || (CurrentUser.Id.HasValue && CurrentUser.Id == Model.Post.CreatorId))
+                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Posts.Delete))
                                                 {
```
Comment (maliming) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Users are not allowed to delete their own Posts?
https://github.com/abpframework/abp/blob/b0adb9ff09e8fe1db495adf06cf544478f7daf68/modules/blogging/src/Volo.Blogging.Application/Volo/Blogging/Posts/PostAuthorizationHandler.cs#L38

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 2]
```diff
@@ -57,14 +57,14 @@
                                                 <i class="fa fa-comment"></i> @L["CommentWithCount", @Model.CommentCount]
 
 
-                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Posts.Update))
+                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Posts.Update) || (CurrentUser.Id.HasValue && CurrentUser.Id == Model.Post.CreatorId))
```
Comment (maliming) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: `(CurrentUser.Id.HasValue && CurrentUser.Id == Model.Post.CreatorId)` can be `CurrentUser.Id == Model.Post.CreatorId`

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 3]
```diff
@@ -160,15 +160,15 @@
...
+                                        @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Delete))
                                         {
                                             <span class="seperator">|</span>
                                             <a href="#" class="tag deleteLink" data-deleteid="@commentWithRepliesDto.Comment.Id">
                                                 <i class="fa fa-trash" aria-hidden="true"></i> @L["Delete"]
                                             </a>
                                         }
 
-                                        @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
+                                        @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id.HasValue && CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
```
Comment (maliming) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Can be `CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId`

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 4]
```diff
@@ -200,7 +200,7 @@
                                             </div>
                                         </div>
                                     }
-                                    @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
+                                    @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id.HasValue && CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
```
Comment (maliming) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Can be `CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId`

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 5]
```diff
@@ -238,15 +238,15 @@
...
+                                                    @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Delete))
                                                     {
                                                         <span class="seperator">|</span>
                                                         <a href="#" class="tag deleteLink" data-deleteid="@reply.Id">
                                                             <i class="fa fa-trash" aria-hidden="true"></i> @L["Delete"]
                                                         </a>
                                                     }
 
-                                                    @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
+                                                    @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id.HasValue && CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
```
Comment (maliming) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Can be `CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId`

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 6]
```diff
@@ -276,7 +276,7 @@
                                                         </div>
                                                     </div>
                                                 }
-                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
+                                                @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Update) || (CurrentUser.Id.HasValue && CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
```
Comment (maliming) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Can be `CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId`

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml (hunk 1 above)
Comment (gdlcf88) [reply modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: > Users are not allowed to delete their own Posts?
> https://github.com/abpframework/abp/blob/b0adb9ff09e8fe1db495adf06cf544478f7daf68/modules/blogging/src/Volo.Blogging.Application/Volo/Blogging/Posts/PostAuthorizationHandler.cs#L38

Sorry, my branch missed this commit:
https://github.com/abpframework/abp/commit/b0adb9ff09e8fe1db495adf06cf544478f7daf68#diff-e461a1b98fc07088163606c19001917c

Comment (maliming) [reply modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: You can merge `dev `into your branch.

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 7]
```diff
@@ -238,15 +238,15 @@
                                                             <i class="fa fa-reply" aria-hidden="true"></i> @L["Reply"]
                                                         </a>
                                                     }
-                                                    @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Delete) || (CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
```
Comment (yekalkan) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Why did you remove `CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId`?

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml [hunk 8]
```diff
@@ -160,15 +160,15 @@
                                             </a>
                                         }
 
-                                        @if (await Authorization.IsGrantedAsync(BloggingPermissions.Comments.Delete) || (CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId))
```
Comment (yekalkan) [inline modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: Why did you remove `CurrentUser.Id == commentWithRepliesDto.Comment.CreatorId`?

File: modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml (hunk 7 above)
Comment (gdlcf88) [reply modules/blogging/src/Volo.Blogging.Web/Pages/Blogs/Posts/Detail.cshtml]: I forgot to merge the latest branch since the creator check was added into AuthorizationHandler 4 days ago. 😅
